deploy-board/deploy_board/webapp/env_views.py

```diff
@@ -784,9 +789,23 @@ def post_create_env(request):
     clone_env_name = data.get("clone_env_name")
     clone_stage_name = data.get("clone_stage_name")
     description = data.get('description')
+    
     if clone_env_name and clone_stage_name:
-        common.clone_from_stage_name(request, env_name, stage_name, clone_env_name,
-                                     clone_stage_name, description)
+        try:
+            external_id = environs_helper.create_identifier_for_new_stage(request, env_name, stage_name)
+            common.clone_from_stage_name(request, env_name, stage_name, clone_env_name,
+                                        clone_stage_name, description, external_id)
+        except Exception as detail:
+            if external_id:
+                try:
+                    environs_helper.delete_nimbus_identifier(request, external_id)
+                except:
+                    message = 'Also failed to delete Nimbus identifier {}. Please verify that identifier no longer exists, Error Message: {}'.format(external_id, detail)
+                    log.error(message)
+            else:
+                message = 'Failed to create identifier for {}/{}: {}'.format(env_name, stage_name, detail)
+                messages.add_message(request, messages.ERROR, message)
+        raise detail
```

Comment:
Hi @rfleur01 @tylerwowen, looks like this `raise` is used outside of the `except` block. Is this intentional or am I missing something?

Reply:
fix is here: https://github.com/pinterest/teletraan/pull/1098